openpype/plugins/publish/collect_frame_range_data.py

```diff
@@ -0,0 +1,67 @@
+import pyblish.api
+import clique
+
+
+class CollectFrameRangeData(pyblish.api.InstancePlugin):
+    """Collect Frame Range data.
+    """
+
+    order = pyblish.api.CollectorOrder + 0.491
+    label = "Collect Frame Range Data"
+    families = ["plate", "pointcache",
+                "vdbcache", "online",
+                "render"]
+    hosts = ["traypublisher"]
+    img_extensions = ["exr", "dpx", "jpg", "jpeg", "png", "tiff", "tga",
+                      "gif", "svg"]
+    video_extensions = ["avi", "mov", "mp4"]
+
+    def process(self, instance):
+        repres = instance.data.get("representations")
+        asset_data = None
+        if repres:
+            first_repre = repres[0]
+            ext = first_repre["ext"].replace(".", "")
+            if not ext or ext.lower() not in self.img_extensions:
+                self.log.warning(f"Cannot find file extension "
+                                 " in representation data")
+                return
+            if ext in self.video_extensions:
+                self.log.info("Collecting frame range data"
+                              " not supported for video extensions")
+                return
+
+            files = first_repre["files"]
+            repres_file = clique.assemble(
+                files, minimum_items=1)[0][0]
+            repres_frames = [frames for frames in repres_file.indexes]
+            last_frame = len(repres_frames) - 1
+            asset_data = {
+                "frameStart": repres_frames[0],
+                "frameEnd": repres_frames[last_frame],
+            }
+
+        else:
+            self.log.info("No representation data.. "
+                          "\nUse Asset Entity data instead")
+            asset_doc = instance.data.get("assetEntity")
+            if instance.data.get("frameStart") is not None or not asset_doc:
+                self.log.debug("Instance has no asset entity set."
+                               " Skipping collecting frame range data.")
+                return
+            asset_data = asset_doc["data"]
```

Comment:
I think we might want to put this fallback behavior behind a setting - I would want my image sequences to get their frame range collected (the if-branhc). But having an Alembic file suddenly getting the frame range of the asset doesn't always feel right to me (especially at the times that it's basically wrong). Maybe this fallback could be toggled by the artist in the publisher, like "Set asset frame range if no frame range detected in source files" or allowing an admin to enable/disable this fallback for specific extensions or alike?

Reply:
I agree that some specific files like those in the pointcache family are not quite logical for using asset_doc data. 
Maybe we can add settings to eanble/disable the fallback for pointcache family and vdb cache (or any non-image type) . @BigRoy

Reply:
I prefer explicit over implicit so on our end we'd likely disable this always. I guess the exact implementation when to "fallback to asset data" would be up to others?